diff --git a/meson.build b/meson.build
index a3c3430..bef39d6 100644
--- a/meson.build
+++ b/meson.build
@@ -393,7 +393,15 @@ endif
 
 if posix
     path_source = files('osdep/path-unix.c')
+    if host_machine.system() == 'darwin'
+    have_tvos = cc.get_define('TARGET_OS_TV',
+        prefix: '#include <TargetConditionals.h>') == '1'
+endif
+if have_tvos
+    subprocess_source = files('osdep/subprocess-dummy.c')
+else
     subprocess_source = files('osdep/subprocess-posix.c')
+endif
     sources += files('input/ipc-unix.c',
                      'osdep/poll_wrapper.c',
                      'osdep/terminal-unix.c',
diff --git a/video/out/hwdec/hwdec_ios_gl.m b/video/out/hwdec/hwdec_ios_gl.m
index 633cc3d..c85b444 100644
--- a/video/out/hwdec/hwdec_ios_gl.m
+++ b/video/out/hwdec/hwdec_ios_gl.m
@@ -90,7 +90,8 @@ static int mapper_init(struct ra_hwdec_mapper *mapper)
 
     for (int n = 0; n < p->desc.num_planes; n++) {
         p->desc.planes[n] = find_la_variant(mapper->ra, p->desc.planes[n]);
-        if (!p->desc.planes[n] || p->desc.planes[n]->ctype != RA_CTYPE_UNORM) {
+        //if (!p->desc.planes[n] || p->desc.planes[n]->ctype != RA_CTYPE_UNORM) {
+        if (!p->desc.planes[n]) {
             MP_ERR(mapper, "Format unsupported.\n");
             return -1;
         }
