diff --git a/src/gpu/utils.c b/src/gpu/utils.c
index 40ca84d..9a8fb91 100644
--- a/src/gpu/utils.c
+++ b/src/gpu/utils.c
@@ -142,7 +142,7 @@ pl_gpu pl_gpu_finalize(struct pl_gpu_t *gpu)
 
     // Verification
     pl_assert(gpu->limits.max_tex_2d_dim);
-    pl_assert(gpu->limits.max_variable_comps || gpu->limits.max_ubo_size);
+    //pl_assert(gpu->limits.max_variable_comps || gpu->limits.max_ubo_size);
     pl_assert(gpu->limits.max_ubo_size    <= gpu->limits.max_buf_size);
     pl_assert(gpu->limits.max_ssbo_size   <= gpu->limits.max_buf_size);
     pl_assert(gpu->limits.max_vbo_size    <= gpu->limits.max_buf_size);
