diff --git a/Common/MVKOSExtensions.mm b/Common/MVKOSExtensions.mm
index 93025f2b..32531fc4 100644
--- a/Common/MVKOSExtensions.mm
+++ b/Common/MVKOSExtensions.mm
@@ -23,6 +23,7 @@
 #include <mach/mach_time.h>
 #include <mach/task.h>
 #include <os/proc.h>
+#include <sys/sysctl.h>
 
 #import <Foundation/Foundation.h>
 
@@ -113,6 +114,11 @@ uint64_t mvkGetSystemMemorySize() {
 	return 0;
 #endif
 #if MVK_TVOS
+	uint64_t host_memsize = 0;
+	size_t size = sizeof(host_memsize);
+	if (sysctlbyname("hw.memsize", &host_memsize, &size, NULL, 0) == KERN_SUCCESS) {
+		return host_memsize;
+	}
 	return 0;
 #endif
 }
